Add multiplex section to QC report

allyhawkins opened the pull request:
Related to https://github.com/AlexsLemonade/scpca-nf/issues/173, this PR adds information related to multiplexed libraries to the QC report. I've gone ahead and added the following: 

- A warning at the beginning of the report stating that the library is multiplexed and contains data for multiple samples. I also included all multiplexed samples that can be found in the library. One thing I thought of that would be helpful is adding a link to the docs for what is a multiplexed sample, but we don't actually have that link yet. 
- `Multiplex` to the list of modalities in the `Raw Sample Metrics` table 
- Some general stats of the HTO library to the `Raw Sample Metrics`, mirroring what was present for CITE-seq 
- A multiplex child report with similar tables to what was present for the CITE-seq libraries that summarizes some information about the HTO's such as total reads and % reads mapped to HTOs. I also added a table with each HTO and then the mean UMI per cell and percent of cells detected (taking the `rowData` from the `altExp`). 
- The last table that I added to the QC report was the multiplexing sample calls that includes a summary of calls for each sample across each method, including the total `NA` for each method. 

For the most part I mirrored what was present in the CITE-seq QC reports, but let me know if you think anything is either missing or maybe should not be reported on the report. 
Also do we want to include any further information besides just the table of sample calls for the demultiplexing? The only other thing that I think could _maybe_ be relevant are the UMAPs coloring by sample id for each method, but wasn't quite sure that was necessary to go that deep. 

I've also included an example of the rendered report for reference when reviewing.
[SCPCL000533_qc_report.html.zip](https://github.com/AlexsLemonade/scpcaTools/files/8766289/SCPCL000533_qc_report.html.zip)

jashapiro commented on inst/rmd/multiplex_qc.Rmd:
````diff
@@ -0,0 +1,91 @@
+**Note:** We have reported the demultiplexed sample calls for each of the above mentioned algorithms, but the multiplexed library has not been separated into individual samples. 
+
+```{r}
+# grab columns that have demuxing results as not all methods could be present
+# e.g. vireo is only used if a matching bulk RNA seq library is there, 
+# otherwise those calls will not be present
+demux_methods <- c("hashedDrops_sampleid", "HTODemux_sampleid", "vireo_sampleid")
+demux_columns <- colnames(colData(filtered_sce)) %in% demux_methods
````
Thank you for this comment! I was confused why you didn't just choose the columns directly, and then I read the comment that I had glanced past the first time through.

I might add a quick check that at least one of the columns is present, so we can print something intelligent if for some reason demux calls were not made.

jashapiro commented on inst/rmd/multiplex_qc.Rmd:
```diff
@@ -0,0 +1,91 @@
+                            full_width = FALSE,
+                            position = "left",
+                            ) |>
+  kableExtra::column_spec(2:3, monospace = TRUE) 
```
Looking at the results, the last column was not monospaced, but rather than just changing 3 to 4 I made it a bit more flexible:
```suggestion
  kableExtra::column_spec(2:ncol(demux_calls), monospace = TRUE) 
```

jashapiro reviewed (commented):
This looks great! Thanks for handling it.

I noticed a couple of little things in the report, including a tiny formatting thing, but also that the sample ID in the initial table isn't complete, which we should do something about. There are a number of ways we could go about that, but I'll leave to you to decide which to do!

allyhawkins commented on inst/rmd/qc_report.rmd:
```diff
@@ -72,10 +72,30 @@ has_cite <- "CITEseq" %in% altExpNames(filtered_sce)
+has_multiplex <- "cellhash" %in% altExpNames(filtered_sce)
+if(has_multiplex){
+  modalities <- c(modalities, "Multiplex")
+  multiplex_samples <- paste(metadata(filtered_sce)$sample_id, collapse = ", ")
```
What do you mean it only shows a single sample? We process a single QC report for each library and so we actually are showing the single library ID here and passing that in as the `sample_id` and not the `scpca_sample_id`. So I think it makes sense to show just the single library ID here. Perhaps we should update this so that we continue to pass in the library ID , but have both the library ID and the sample ID (since we can now grab the sample ID from the metadata of the SCE) displayed in the table?

jashapiro replied:
Oh you are right! I was confused by the "sample" label. We should probably change the name of that param & variable to reflect that they are library ids. Having the actual sample id in the table as well seems like it would be a good idea!

allyhawkins replied:
Yes I think that all sounds good! Will make those changes throughout to reflect that it's a library ID. We will also need to remember to update the script that generates the QC report in `scpca-nf` when we change the parameter name.

allyhawkins commented:
Thanks for taking a look at this @jashapiro. I went through your comments and added the following: 

- I changed the warning so it has a bulleted list of sample ids for multiplexed samples rather than the comma separated list
- I added a check for the demultiplexing column names and print out a warning statement if there are no demultiplexing calls. I tried to change this color to be something other than yellow, but for some reason the other one that I tried didn't work as expected (`alert-primary`) and only the `alert-warning` created the colored block. 
- I made updates throughout to clarify between the use of sample and library id. I now have the QC report requiring a library ID as the input, and explicitly saying that as the name of the argument rather than `sample_name` in `generate_qc_report()`. I then changed the parameter name for the template notebook to be `library_id` rather than `sample`. I also included both the library ID and the sample ID in the `sample_information` table as mentioned in https://github.com/AlexsLemonade/scpcaTools/pull/105#discussion_r881910694. If you are okay with this, then I'll file an issue to make the change in `scpca-nf` to use the correct argument name when running `scpcaTools::generate_qc_report` once we tag this release and update to using the new docker container. 

Here's an example of the updated report with these changes. 
[SCPCL000533_qc_report.html.zip](https://github.com/AlexsLemonade/scpcaTools/files/8774213/SCPCL000533_qc_report.html.zip)
